File: src/Servidor/Transaccional/Comandos/Comandos/ComandosDepartamento/ComandoConsultarDepartamentoPorId.cs
using System.Configuration;
using NLog;
using Trascend.Bolet.AccesoDatos.Contrato;
using Trascend.Bolet.AccesoDatos.Fabrica;
using Trascend.Bolet.ObjetosComunes.Entidades;
using System;

namespace Trascend.Bolet.Comandos.Comandos.ComandosDepartamento
{
    public class ComandoConsultarDepartamentoPorId : ComandoBase<Departamento>
    {
        private static Logger logger = LogManager.GetCurrentClassLogger();
        private Departamento _departamento;

        /// <summary>
        /// Constructor predeterminado
        /// </summary>
        /// <param name="departamento"></param>
        public ComandoConsultarDepartamentoPorId(Departamento departamento)
        {
            this._departamento = departamento;
        }

        /// <summary>
        /// Método que ejecuta el comando
        /// </summary>
        public override void Ejecutar()
        {
            try
            {
                #region trace
                if (ConfigurationManager.AppSettings["Ambiente"].ToString().Equals("Desarrollo"))
                    logger.Debug("Entrando al metodo {0}", (new System.Diagnostics.StackFrame()).GetMethod().Name);
                #endregion

                IDaoDepartamento dao = FabricaDaoBase.ObtenerFabricaDao().ObtenerDaoDepartamento();
                this.Receptor = new Receptor<Departamento>(dao.ObtenerPorId(this._departamento.Id));

                #region trace
                if (ConfigurationManager.AppSettings["Ambiente"].ToString().Equals("Desarrollo"))
                    logger.Debug("Saliendo del metodo {0}", (new System.Diagnostics.StackFrame()).GetMethod().Name);
                #endregion
            }
            catch (ApplicationException ex)
            {
                logger.Error(ex.Message);
                throw ex;
            }
        }
    }
}

File: trunk/src/Servidor/Transaccional/Comandos_Fac/Comandos/ComandosDepartamento/ComandoConsultarDepartamentoPorId.vb
Imports System.Configuration
Imports NLog
Imports Diginsoft.Bolet.AccesoDatos.Contrato
Imports Diginsoft.Bolet.AccesoDatos.Fabrica
Imports Diginsoft.Bolet.ObjetosComunes.Entidades

Namespace Comandos.ComandosDepartamento
    Public Class ComandoConsultarDepartamentoPorId
        Inherits ComandoBase(Of Departamento)
        Private Shared logger As Logger = LogManager.GetCurrentClassLogger()
        Private _departamento As Departamento

        ''' <summary>
        ''' Constructor predeterminado
        ''' </summary>
        ''' <param name="departamento"></param>
        Public Sub New(ByVal departamento As Departamento)
            Me._departamento = departamento
        End Sub

        ''' <summary>
        ''' Método que ejecuta el comando
        ''' </summary>
        Public Overrides Sub Ejecutar()
            Try
                '#Region "trace"
                If ConfigurationManager.AppSettings("Ambiente").ToString().Equals("Desarrollo") Then
                    logger.Debug("Entrando al metodo {0}", (New System.Diagnostics.StackFrame()).GetMethod().Name)
                End If
                '#End Region

                Dim dao As IDaoDepartamento = FabricaDaoBase.ObtenerFabricaDao().ObtenerDaoDepartamento()
                Me.Receptor = New Receptor(Of Departamento)(dao.ObtenerPorId(Me._departamento.Id))

                '#Region "trace"
                If ConfigurationManager.AppSettings("Ambiente").ToString().Equals("Desarrollo") Then
                    logger.Debug("Saliendo del metodo {0}", (New System.Diagnostics.StackFrame()).GetMethod().Name)
                    '#End Region
                End If
            Catch ex As ApplicationException
                logger.[Error](ex.Message)
                Throw ex
            End Try
        End Sub
    End Class
End Namespace